File: src/platform/web/backend/mcp.ts
import { definePlatformCommandHandlers } from "./command-handler";

export const mcpCommandHandlers = definePlatformCommandHandlers({
  mcp_settings_read: () => ({
    settings: { enabled: false, host: "127.0.0.1", port: 14243, token: null },
    status: { supported: false, running: false, endpoint: null, last_error: null },
  }),
  mcp_settings_write: () => { throw new Error("MCP server requires the desktop application."); },
});

File: crates/ironwing-core/src/ipc/mcp.rs
//! Desktop MCP listener configuration. Secrets never appear in server status/events.
#[cfg_attr(feature = "typescript", derive(specta::Type))]
#[derive(Clone, serde::Serialize, serde::Deserialize, PartialEq, Eq)]
pub struct McpSettings {
    pub enabled: bool,
    pub host: String,
    pub port: u16,
    pub token: Option<String>,
}
impl Default for McpSettings {
    fn default() -> Self {
        Self {
            enabled: false,
            host: "127.0.0.1".into(),
            port: 14243,
            token: None,
        }
    }
}
#[cfg_attr(feature = "typescript", derive(specta::Type))]
#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
pub struct McpStatus {
    pub supported: bool,
    pub running: bool,
    pub endpoint: Option<String>,
    pub last_error: Option<String>,
}
#[cfg_attr(feature = "typescript", derive(specta::Type))]
#[derive(Clone, serde::Serialize, serde::Deserialize)]
pub struct McpSettingsResult {
    pub settings: McpSettings,
    pub status: McpStatus,
}
